File: app/src/main/java/com/example/networkaplication/webview/WebViewFragment.java
package com.example.networkaplication.webview;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebView;

import com.example.networkaplication.R;

import java.util.Objects;

public class WebViewFragment extends Fragment {

    private WebView webView;
    private String url;
    private static final String URL = "URL";

    public static WebViewFragment newInstance(String url) {
        WebViewFragment webViewFragment = new WebViewFragment();
        Bundle bundle = new Bundle();
        bundle.putString(URL, url);
        webViewFragment.setArguments(bundle);

        return webViewFragment;
    }

    @Override
    public void onStart() {
        super.onStart();
        setHasOptionsMenu(true);
        Objects.requireNonNull(getActivity()).setTitle("WebView");
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getArguments() != null)
            url = getArguments().getString(URL);
    }

    @SuppressLint("SetJavaScriptEnabled")
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {

        View rootView = inflater.inflate(R.layout.web_view_fragment, container, false);

        webView = rootView.findViewById(R.id.web_view);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.setWebViewClient(new MyWebViewClient());
        webView.loadUrl(url);

        return rootView;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId()) {
            case android.R.id.home:
                if (webView.canGoBack())
                    webView.goBack();
                else
                    Objects.requireNonNull(getActivity()).onBackPressed();
        }

        return super.onOptionsItemSelected(item);
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public WebView getWebView() {
        return webView;
    }
}

File: app/src/main/java/com/example/networkaplication/webview/WebViewFragment.kt
package com.example.networkaplication.webview

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.webkit.WebView
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity

import com.example.networkaplication.R

import java.util.Objects

class WebViewFragment : Fragment() {

    var webView: WebView? = null
        private set
    private var url: String? = null

    override fun onStart() {
        super.onStart()
        setHasOptionsMenu(true)
        Objects.requireNonNull<FragmentActivity>(activity).setTitle("WebView")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (arguments != null)
            url = arguments!!.getString(URL)
    }

    @SuppressLint("SetJavaScriptEnabled")
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        val rootView = inflater.inflate(R.layout.web_view_fragment, container, false)

        webView = rootView.findViewById(R.id.web_view)
        webView!!.settings.javaScriptEnabled = true
        webView!!.webViewClient = MyWebViewClient()
        webView!!.loadUrl(url)

        return rootView
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {

        if (item!!.itemId == android.R.id.home) {
            if (webView!!.canGoBack())
                webView!!.goBack()
            else
                Objects.requireNonNull<FragmentActivity>(activity).onBackPressed()
        }

        return super.onOptionsItemSelected(item)
    }

    fun setUrl(url: String) {
        this.url = url
    }

    companion object {
        private val URL = "URL"

        fun newInstance(url: String): WebViewFragment {
            val webViewFragment = WebViewFragment()
            val bundle = Bundle()
            bundle.putString(URL, url)
            webViewFragment.arguments = bundle

            return webViewFragment
        }
    }
}
